src/components/profile/Edit.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import { useDropzone } from 'react-dropzone'
+import { useGlobalContext } from '../../store/global'
+
+const Edit = () => {
+
+  const { user, logout, updateProfile } = useGlobalContext()
+
+	const [pageData, setPageData] = useState({
+		fileSrc: null
+	})
+
+  // react-dropzone receives uploaded images files here
+  // and updates pageData.fileSrc with the blobbed data
+	const onDrop = useCallback(acceptedFiles => {
+		const reader = new FileReader()
+		reader.onload = () => {
+			const res = reader.result
+			setPageData(prevState => ({
+				...prevState,
+				fileSrc: res
+			}))
+		}
+		if (acceptedFiles[0]) {
+			reader.readAsDataURL(acceptedFiles[0])	
+		}
+	}, [])
```

Comment:
I think you're missing some deps for the `useCallback` hook.

Reply:
I honestly get a little confused with this sometimes, and often depend on a linter or create-react-app to warn me about it. I added what i think should be the correct dependencies here. Please correct if I still have it wrong.